11_FFTBloom/main.cpp

```diff
@@ -449,80 +473,73 @@ class FFTBloomApp final : public application_templates::MonoDeviceApplication, p
+		smart_refctd_ptr<IGPUPipelineLayout> pipelineLayout;
 		{
-			auto sampler = createSampler(ISampler::E_TEXTURE_CLAMP::ETC_MIRROR);
-			smart_refctd_ptr<IGPUSampler> samplers[Channels];
-			std::fill_n(samplers, Channels, sampler);
-			IGPUDescriptorSetLayout::SBinding bnd =
+			IGPUDescriptorSetLayout::SBinding bnd[4] =
 			{
-				IDescriptorSetLayoutBase::SBindingBase(),
-				0u,
-				IDescriptor::E_TYPE::ET_COMBINED_IMAGE_SAMPLER,
-				IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
-				IShader::E_SHADER_STAGE::ESS_COMPUTE,
-				Channels,
-				samplers
+				// Kernel FFT and Image FFT read from a single Image
+				{
+					IDescriptorSetLayoutBase::SBindingBase(),
+					0u,
+					IDescriptor::E_TYPE::ET_SAMPLED_IMAGE,
+					IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+					IShader::E_SHADER_STAGE::ESS_COMPUTE,
+					1u,
+					nullptr
+				},
+				// Sampler: First Axis FFT (image) and convolution use a mirror-sampler, First Axis FFT (kernel) uses a clamp to border
+				{
+					IDescriptorSetLayoutBase::SBindingBase(),
+					1u,
+					IDescriptor::E_TYPE::ET_SAMPLER,
+					IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+					IShader::E_SHADER_STAGE::ESS_COMPUTE,
+					1u,
+					nullptr
+				},
+				// Storage Image: Normalization binds a texture array, First Axis IFFT binds a single image
+				{
+					IDescriptorSetLayoutBase::SBindingBase(),
+					2u,
+					IDescriptor::E_TYPE::ET_STORAGE_IMAGE,
+					IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+					IShader::E_SHADER_STAGE::ESS_COMPUTE,
+					1,
+					nullptr
+				},
+				// Convolution binds a texture array. Trying to have this in same binding slot as image would be cool but we lose the ability to write the descriptor set only once
+				{
+					IDescriptorSetLayoutBase::SBindingBase(),
+					3u,
+					IDescriptor::E_TYPE::ET_SAMPLED_IMAGE,
+					IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+					IShader::E_SHADER_STAGE::ESS_COMPUTE,
+					1,
+					nullptr
+				}
 			};
 
-			lastAxisFFT_convolution_lastAxisIFFTPipelineLayout = createPipelineLayout({ &bnd, 1 });
+			pipelineLayout = createPipelineLayout({ bnd, 4 });
 		}
 
-		smart_refctd_ptr<IGPUPipelineLayout> imageFirstAxisIFFTPipelineLayout;
-		{
-			IGPUDescriptorSetLayout::SBinding bnd =
-			{
-				IDescriptorSetLayoutBase::SBindingBase(),
-				0u,
-				IDescriptor::E_TYPE::ET_STORAGE_IMAGE,
-				IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
-				IShader::E_SHADER_STAGE::ESS_COMPUTE,
-				1,
-				nullptr
-			};
+		// Create descriptor set
+		const IGPUDescriptorSetLayout* descriptorSetLayout = pipelineLayout->getDescriptorSetLayout(0);
+		smart_refctd_ptr<IDescriptorPool> descriptorSetPool = m_device->createDescriptorPoolForDSLayouts(IDescriptorPool::ECF_NONE, { &descriptorSetLayout, 1 });
+		uint32_t dsCreated = descriptorSetPool->createDescriptorSets({ &descriptorSetLayout, 1 }, &m_descriptorSet);
+		if (dsCreated != 1)
```

Comment:
it would really simplify your code to create the descriptor set with the Asset Converter from an ICPUDescriptorSet with semi-unpopulated bindings https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/144#discussion_r1836586236

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/144#discussion_r1836575814

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/144#discussion_r1836574984

Reply:
Oh yeah this one was pending as well